### DISCUSSION TITLE
Visualizer cumulative subscriptions plot shows negative values for default broker

### DISCUSSION THREAD

**Discussion by jecollins** [powertac/powertac-server#1173] [General]
It apparently fails to account for the fact that at start-of-game the default broker has all the subscriptions, while all the other brokers start with zero subscriptions.

**Comment by erikkemperman** [comment]
I've looked into this -- but I am not sure what is going wrong...

> at start-of-game the default broker has all the subscriptions

This is because of it being the only participant during boot, correct?

 Couple of things that I notice:

- The default broker looks for `CustomerBootstrapData` messages and accounts for subscriptions based on those messages at the beginning of a sim. 
- The viz also listens for those messages, but at the moment only accounts for net usage, not subscriptions.
- When I make the viz account for subscriptions as well, say that it starts with default broker having N subscriptions, then I notice that after the first timeslot it gains _another_ N (precisely). Not sure why that would happen -- perhaps the default broker is remembering its subscriptions from boot, whilst the customers are not and thus immediately seek new ones?
- These `CustomerBootstrapData` messages are not recorded in sim state logs, so they don't come by during replay. This is likely a separate issue, but did you see negative numbers of subs during an actual game or during replay?

**Comment by jecollins** [comment]
Good question. This was in a replay. In a real game, all brokers start at zero, the DB gets 172000 in the 2nd timeslot, and then loses most of them in the 3rd timeslot. I'm not sure why it does not lose all of them, I'll look into it. But 172000 seems too high by more than a factor of 2. So I re-loaded the page, and now the DB starts with 57000 in the first ts, and drops to about 7000 in the second. Strange.
